settings.php

```diff
@@ -7,56 +7,64 @@
 
-        // Add settings page A to the folder
-        $global_settings = new admin_settingpage('tool_paymentplugin_globalsettings', get_string('gsettings', 'tool_paymentplugin'));
-        $ADMIN->add('tool_paymentplugin_folder', $global_settings);
+        // Create a category in the admin tree
+        $paymentplugincat = new admin_category('tool_paymentplugin_folder', get_string('pluginname', 'tool_paymentplugin'), false);
+        $paymentplugincat->add('tool_paymentplugin_folder', $global_settingsA);
+        $paymentplugincat->add('tool_paymentplugin_folder', $global_settingsB);
+        $paymentplugincat->add('tool_paymentplugin_folder', $global_settingsC);
+        // Add the category to the tree
+        $ADMIN->add('tools', $paymentplugincat);
 
-        // Add settings page B to the folder
-        $global_settingsB = new admin_settingpage('tool_paymentplugin_globalsettingsB', get_string('GsettingsB', 'tool_paymentplugin'));
-        $ADMIN->add('tool_paymentplugin_folder', $global_settingsB);
 
-        // Add settings page C to the folder
-        $global_settingsC = new admin_settingpage('tool_paymentplugin_globalsettingsC', get_string('GsettingsC', 'tool_paymentplugin'));
-        $ADMIN->add('tool_paymentplugin_folder', $global_settingsC);
-
-        /**
-         * 
-         * Global Settings Page A Setup
-         * 
-         * 
-         */
-        // Add Heading
-        $global_settings->add(new admin_setting_heading('tool_paymentplugin_globalsettings/heading', get_string('gsettingsheader', 'tool_paymentplugin'), 
-            get_string('gsettingsdesc', 'tool_paymentplugin')));
-
-        // Add checkboxes
-        $global_settings->add(new admin_setting_configcheckbox('tool_paymentplugin_globalsettings/disablePurchases', get_string('gsettingsdisablepurchase', 'tool_paymentplugin'), 
-            get_string('gsettingsdisablepurchasedesc', 'tool_paymentplugin'), 0));   
-        $global_settings->add(new admin_setting_configcheckbox('tool_paymentplugin_globalsettings/checkbox2', get_string('gsettingscheck2', 'tool_paymentplugin'), 
-            get_string('gsettingscheck2desc', 'tool_paymentplugin'), 0));
-        
-        // Add multi select
+        // Create settings
+        $heading = new admin_setting_heading('tool_paymentplugin_GsettingsA/heading', $global_settingsA->visiblename, 
+            get_string('GsettingsA_desc', 'tool_paymentplugin'));
+
+        $disableallcheck = new admin_setting_configcheckbox('tool_paymentplugin_GsettingsA/disablePurchases', get_string('GsettingsA_disablePurchase', 'tool_paymentplugin'), 
+            get_string('GsettingsA_disablePurchase_desc', 'tool_paymentplugin'), 0);
+
+        $checkbox2 = new admin_setting_configcheckbox('tool_paymentplugin_GsettingsA/checkbox2', get_string('GsettingsA_check2', 'tool_paymentplugin'), 
+            get_string('GsettingsA_check2_desc', 'tool_paymentplugin'), 0);
```

Comment:
All the changes look good (I haven't worked enough with navigation API to be able to spot problems, but it looks fine for me)
Only thing I'm worried about is the language string names like `GsettingsA`, but they seem like placeholder names so I'm guessing you'll change them down the line anyway

Reply:
Ignore what I just said, tested out this branch from my side and got a lot of errors because all the language strings beginning with `GsettingsA` don't exist in tool_paymentplugin.php

Reply:
Fixed it, sorry about that, could have sworn I fixed that.